src/main/features/claim/routes/pay.ts

```diff
@@ -85,30 +85,30 @@ export default express.Router()
         throw new Error('No amount entered, you cannot pay yet')
       }
 
-      const paymentId = draft.document.claimant.payment.id
+      const paymentRef = draft.document.claimant.payment.reference
 
-      if (paymentId) {
+      if (paymentRef) {
         const payClient: PayClient = await getPayClient()
-        const payment: Payment = await payClient.retrieve(user, paymentId)
-        switch (payment.state.status) {
-          case 'success':
+        const paymentResponse: PaymentResponse = await payClient.retrieve(user, paymentRef)
+        switch (paymentResponse.status) {
+          case 'Success':
             return res.redirect(Paths.finishPaymentReceiver.evaluateUri({ externalId: draft.document.externalId }))
         }
       }
       const feeOutcome: FeeOutcome = await FeesClient.calculateFee(event, amount, channel)
       const payClient: PayClient = await getPayClient()
-      const amountInPennies: number = MoneyConverter.convertPoundsToPennies(feeOutcome.amount)
-      const payment: PaymentResponse = await payClient.create(
+      const fees: Fees = new Fees(feeOutcome.amount, feeOutcome.code, feeOutcome.version)
+      const feesArray: Fees[] = [fees]
```

Comment:
Please inline it, no value added by extracting this local variable.

Reply:
Done